### PR TITLE
[Impeller] Ensure that reused textures are cleared before getting sampled by backdrop textures

### PR DESCRIPTION
**Pull request by bdero**
Fix for https://github.com/flutter/flutter/issues/135053.

I'm going to try and optimize out this case later, since Wondrous is inadvertently paying for a backdrop filter that contributes nothing to the final image.

Wondrous with all fixes applied:

https://github.com/flutter/engine/assets/919017/37b49b77-b0ec-42d4-964a-c61b9a2cca8f

### CODE REVIEW

File: impeller/entity/entity_pass.cc [hunk 1]
```diff
@@ -537,6 +537,7 @@ EntityPass::EntityResult EntityPass::GetEntityForElement(
       // The subpass will need to read from the current pass texture when
       // rendering the backdrop, so if there's an active pass, end it prior to
       // rendering the subpass.
+      pass_context.GetRenderPass(pass_depth);
```
Comment (jonahwilliams) [inline impeller/entity/entity_pass.cc:540]: I'm not sure I fully understand this case 🤔

Comment (bdero) [reply impeller/entity/entity_pass.cc]: If the very first thing we render in an EntityPass is a subpass that happens to have a backdrop filter, than that backdrop filter will end up rendering the raw, uncleared texture that came straight out of the texture cache. By calling `pass_context.GetRenderPass` here, we force at least one pass to get created with the correct config before that sampling happens, which will clear the texture.

Added this to a comment inline.

Comment (jonahwilliams) [reply impeller/entity/entity_pass.cc]: Oh, I see - so we're attaching the texture but we never created any rendering commands to record to it, so we never cleared the values. hmm. Slight flaw in the render target cache design

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Pull request merged**